src/main/groovy/org/pih/warehouse/inventory/CycleCountRequestCommand.groovy

```diff
@@ -0,0 +1,33 @@
+package org.pih.warehouse.inventory
+
+import grails.validation.Validateable
+import org.pih.warehouse.core.Location
+import org.pih.warehouse.product.Product
+
+class CycleCountRequestCommand implements Validateable {
+
+    Location facility
+
+    List<Product> products
+
+    Boolean blindCount
+
+    static constraints = {
+        products(nullable: true, validator: { List<Product> products, CycleCountRequestCommand obj ->
+            String productCode = null
+            boolean duplicateExists = products.any { Product product ->
+                // FIXME: Most probably, this will have to be modified not to include completed cycle counts
```

Comment:
yeah you'd want to change the findBy to something like a `countByProductAndFacilityAndStatus` (where status not in [COMPLETED, CANCELED])

Reply:
yeah, I left that `FIXME` to remember, as this will be doable after `CycleCount` domain is created and working.